docker/server/entrypoint.sh

```diff
@@ -62,7 +63,11 @@ if [ -n "$(ls /docker-entrypoint-initdb.d/)" ]; then
         exit 1
     fi
 
-    clickhouseclient=( clickhouse-client --multiquery )
+    if [ ! -z "$CLICKHOUSE_PASS" ]; then
+        WITH_PASSWD="--password $CLICKHOUSE_PASS"
```

Comment:
1) password can contain unsafe (for bash) chars
2) PASS is bad name, use full name
3) In most of docker DB entrypoints PASSWORD variable allows to set up a password for a user, not only use that for init.db, so that can be misleading. 
See - https://github.com/docker-library/mysql/blob/master/8.0/docker-entrypoint.sh
https://github.com/docker-library/postgres/blob/master/11/Dockerfile
4) lack of documentation 

So the proper way
1) rename var to CLICKHOUSE_PASSWORD
2) put small xml to users.d subdirectory of the directory where CLICKHOUSE_CONFIG is placed to create a user with desired name / password.
3) Escape password with single quotes, and something like 
```
WITH_PASSWORD=$(printf "-- password '%q'" "$CLICKHOUSE_PASSWORD")
```
4) add documentation to https://github.com/yandex/ClickHouse/blob/master/docker/server/README.md

As an alternative - special env like CLICKHOUSE_INITDB_USER / CLICKHOUSE_INITDB_PASSWORD can be considered,  
 similar to that:
https://github.com/docker-library/mongo/blob/40056ae591c1caca88ffbec2a426e4da07e02d57/4.0/docker-entrypoint.sh#L194